src/OpenLoco/MessageManager.cpp

```diff
@@ -37,9 +40,74 @@ namespace OpenLoco::MessageManager
         call(0x004285BA, regs);
     }
 
+    // 0x00428E0F
+    static void clearActiveMessage()
+    {
+        if (getGameState().activeMessageIndex != MessageId::null)
+        {
+            get(getGameState().activeMessageIndex)->var_C8 = 0xFFFF;
+        }
+        Ui::WindowManager::close(Ui::WindowType::news);
+        getGameState().activeMessageIndex = MessageId::null;
+    }
+
+    // 0x00428DAB
+    static void remove(const MessageId id)
+    {
+        if (getGameState().activeMessageIndex != MessageId::null)
+        {
+            if (id == getGameState().activeMessageIndex)
+            {
+                clearActiveMessage();
+            }
+            else if (enumValue(id) < enumValue(getGameState().activeMessageIndex))
+            {
+                getGameState().activeMessageIndex = static_cast<MessageId>(enumValue(getGameState().activeMessageIndex) - 1);
+            }
+        }
+        numMessages()--;
+        // Move element to end of array
+        if (enumValue(id) < Limits::kMaxMessages - 1)
+        {
+            std::rotate(get(id), get(id) + 1, std::end(rawMessages()));
+        }
+        Ui::WindowManager::invalidate(Ui::WindowType::messages);
+    }
+
+    // 0x0042851C
+    void removeAllSubjectRefs(const uint16_t subject, MessageItemArgumentType type)
+    {
+        for (auto i = numMessages(); i > 0; --i)
+        {
+            auto& message = rawMessages()[i];
+            const auto& descriptor = getMessageTypeDescriptor(message.type);
+            bool hasModified = false;
+            for (auto j = 0; j < 3; ++j)
```

Comment:
instead of magic number, you could either
- make the 3 a constant in Message.h for the count of message/item ids
- or make the array you added in #!381 (`uint16_t itemSubjects[3];`) a std::array and then you can use std::array::size() here.

Reply:
Sadly can't do ranged for as it's walking two different arrays. Yeah could add something for the constant

Reply:
Let's go with the constant, for now, then.